packages/suite/src/components/suite/modals/AccessCoinjoinAccount.tsx

```diff
@@ -0,0 +1,143 @@
+import React, { useState } from 'react';
+import styled from 'styled-components';
+
+import { UserContextPayload } from '@suite-actions/modalActions';
+import { Button, P, Image, variables, H2, H3, Warning } from '@trezor/components';
+import { Modal, Translation } from '@suite-components';
+
+const SmallModal = styled(Modal)`
+    width: 560px;
+`;
+
+const Description = styled(P)`
+    font-size: ${variables.FONT_SIZE.SMALL};
+    color: ${props => props.theme.TYPE_LIGHT_GREY};
+    text-align: center;
+`;
+
+const DescriptionLeft = styled(Description)`
+    text-align: left;
+`;
+
+const ImageContainer = styled.div`
+    display: flex;
+    flex-direction: column;
+    align-items: center;
+    justify-content: center;
+    margin-bottom: 28px;
+`;
+
+const StyledImage = styled(Image)`
+    width: auto;
+    height: 150px;
+`;
+
+const StyledH2 = styled(H2)`
+    margin-top: 12px;
+    margin-bottom: 12px;
+    font-weight: ${variables.FONT_WEIGHT.MEDIUM};
+    font-size: ${variables.FONT_SIZE.H2};
+`;
+
+const StyledH3 = styled(H3)`
+    margin-top: 12px;
+    font-weight: ${variables.FONT_WEIGHT.MEDIUM};
+    font-size: ${variables.FONT_SIZE.H3};
+    text-align: left;
+`;
+
+const PlaceHolder = styled.div`
+    width: 100%;
+    height: 74px;
+    background-color: #ccc;
+`;
+
+type AccessCoinjoinAccountProps = Omit<
+    Extract<UserContextPayload, { type: 'access-coinjoin-account' }>,
+    'type'
+> & {
+    onCancel: () => void;
+};
+
+export const AccessCoinjoinAccount = ({ onCancel, decision }: AccessCoinjoinAccountProps) => {
+    const [step, setStep] = useState('initial');
+
+    const onContinue = () => {
+        switch (step) {
+            case 'initial':
+                return setStep('anonymity-level');
+            case 'anonymity-level':
+                // When returning `true` we successfully completed the process.
+                return decision.resolve(true);
+            // no default
+        }
+    };
+
+    const onBackClick = () => {
+        switch (step) {
+            case 'initial':
+                return decision.resolve(false);
+            case 'anonymity-level':
+                return setStep('initial');
```

Comment:
If steps are an enum, you can just increment/decrement them ;)

Reply:
If all the steps were only rendering something yes, but here some of them are returning decision so it would be possible but I see it would make it more complex.

Reply:
The switch is ok, I meant that for steps that only increment/decrement you could have a shared case with 

```
setStep(currentStep => currentStep + 1);
```

Reply:
I seeee, I like that, what about [d9602a1](https://github.com/trezor/trezor-suite/pull/6454/commits/d9602a1275d3ecf6d223b02443d42aed74b99b65) ?